base/arrayshow.jl

```diff
@@ -415,6 +413,7 @@ _show_empty(io, X) = nothing # by default, we don't know this constructor
 
 # typeinfo aware (necessarily)
 function show(io::IO, X::AbstractArray)
+    ndims(X) == 0 && return show_zero_dim(io, X)
     ndims(X) == 1 && return show_vector(io, X)
```

Comment:
I did this to match the line below... but we could just use dispatch as usual, since `ndims` is in the type info

I _think_ we don't do this for `show_vector` because it's not actually just for `Vector`s. the attached comment is
```
# NOTE: v is not constrained to be a vector, as this function can work with iterables
# in general (it's used e.g. by show(::IO, ::Set))
````

Shall I change `show_zero_dim` to just extend `show`?

Reply:
I think it's good as-is since there's less potential for ambiguities.